### PR TITLE
feat(crons): Add monitor slug to APIs

### PR DESCRIPTION
**Pull request by evanpurkhiser**
* Returns slug as part of the serialized monitor
 * Allows slug to be specified for creation and updates

### CODE REVIEW

**Review by markstory — APPROVED** [review]

File: tests/sentry/api/endpoints/test_organization_monitor_details.py [hunk 1]
```diff
@@ -44,6 +44,16 @@ def test_name(self):
         monitor = Monitor.objects.get(id=monitor.id)
         assert monitor.name == "Monitor Name"
 
+    def test_slug(self):
```
Comment (rjo100) [inline tests/sentry/api/endpoints/test_organization_monitor_details.py:47]: worth adding a test where `slug` is specified but is `None` or `""`, I don't think the regex checker will allow it in `MonitorValidator`. generally we allow `None` values as equivalent to not specifying an optional param

**Review by dashed — APPROVED** [review]

File: src/sentry/api/validators/monitor.py [hunk 2]
```diff
@@ -89,6 +89,7 @@ def validate(self, attrs):
 class MonitorValidator(serializers.Serializer):
     project = ProjectField(scope="project:read")
     name = serializers.CharField()
+    slug = serializers.RegexField(r"^[a-z0-9_\-]+$", max_length=50, required=False)
```
[the remark this answers is not in the document]
Comment (evanpurkhiser) [reply src/sentry/api/validators/monitor.py:92]: There's a lot of inconsistent `slug = serializer.RegexField`

File: tests/sentry/api/endpoints/test_organization_monitor_details.py (hunk 1 above)
Comment (evanpurkhiser) [reply tests/sentry/api/endpoints/test_organization_monitor_details.py:59]: Updated. It fails with 'cannot be blank'

**Review by rjo100 — APPROVED** [review]
